Can't compile index range on alias type

<!-- It is advisable to update all relevant modules using `v outdated` and `v install` -->
**V doctor:**
```
OS: macos, macOS, 11.2.3, 20D91
Processor: 8 cpus, 64bit, little endian, Intel(R) Core(TM) i7-4850HQ CPU @ 2.30GHz
CC version: Apple clang version 12.0.5 (clang-1205.0.22.11)

getwd: /Users/elliot/v/src/github.com/elliotchance/v-bigint
vmodules: /Users/elliot/.vmodules
vroot: /Users/elliot/v
vexe: /Users/elliot/v/v
vexe mtime: 2021-07-05 22:56:24
is vroot writable: true
is vmodules writable: true
V full version: V 0.2.2 9e61321

Git version: git version 2.30.1 (Apple Git-130)
Git vroot status: Error: fatal: not a git repository (or any of the parent directories): .git
.git/config present: false
thirdparty/tcc status: thirdparty-macos-amd64 689c8a02-dirty
```

**What did you do?**
`v -g -o vdbg cmd/v && vdbg bug/slice_index.v`
```v
type Ints = []int

fn (i Ints) slice(to int) Ints {
	return i[0..to]
}

values := Ints([5, 7, 9])

println(values.slice(2))

```

**What did you expect to see?**

[5, 7]


**What did you see instead?**
```
==================
                              ^
/tmp/v/slice_index.14912723080538647682.tmp.c:10101:23: error: expected ';' at end of declaration
        main__Ints _t727 = i, 0, to);
                             ^
                             ;
2 errors generated.
...
==================
(Use `v -cg` to print the entire error message)

builder error: 
==================
C error. This should never happen.

If you were not working with C interop, this is a compiler bug, please report the bug using `v bug file.v`.

https://github.com/vlang/v/issues/new/choose

You can also use #help on Discord: https://discord.gg/vlang

```

--- a/vlib/v/gen/c/index.v
+++ b/vlib/v/gen/c/index.v
@@ -58,7 +58,7 @@ fn (mut g Gen) index_expr(node ast.IndexExpr) {
 }
 
 fn (mut g Gen) range_expr(node ast.IndexExpr, range ast.RangeExpr) {
-	sym := g.table.get_type_symbol(node.left_type)
+	sym := g.table.get_final_type_symbol(node.left_type)
 	if sym.kind == .string {
 		g.write('string_substr(')
 		g.expr(node.left)

--- a/vlib/v/tests/array_of_alias_slice_test.v
+++ b/vlib/v/tests/array_of_alias_slice_test.v
@@ -1,0 +1,12 @@
+type Ints = []int
+
+fn (i Ints) slice(to int) Ints {
+	return i[0..to]
+}
+
+fn test_array_of_alias_slice() {
+	values := Ints([5, 7, 9])
+
+	println(values.slice(2))
+	assert values.slice(2) == Ints([5, 7])
+}

